This change is part of the master branch but was never integrated into the 2.1
branch, which only has a similar but less extensive commit mentioning Fedora 22.

This backport is necessary for a clean cherry-pick of the change adding support
for OpenSSL 1.1.0 upstream.
---
From 802180b9611e816b12b58c279824106514941d53 Mon Sep 17 00:00:00 2001
From: Ivan Romanov <drizt@land.ru>
Date: Sat, 30 Sep 2017 15:45:59 +0500
Subject: [PATCH 2/2] Disable missed openssl cipher suites

Fedora 26 has no them.
---
 unittest/tls/tlsunittest.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/unittest/tls/tlsunittest.cpp b/unittest/tls/tlsunittest.cpp
index 74c0b18..38caf2d 100644
--- unittest/tls/tlsunittest.cpp
+++ unittest/tls/tlsunittest.cpp
@@ -68,8 +68,12 @@ void TLSUnitTest::testCipherList()
 	QVERIFY( cipherList.contains("TLS_DHE_RSA_WITH_AES_128_CBC_SHA") );
 	QVERIFY( cipherList.contains("TLS_DHE_DSS_WITH_AES_128_CBC_SHA") );
 	QVERIFY( cipherList.contains("TLS_RSA_WITH_AES_128_CBC_SHA") );
-	QVERIFY( cipherList.contains("TLS_RSA_WITH_RC4_128_SHA") );
-	QVERIFY( cipherList.contains("TLS_RSA_WITH_RC4_128_MD5") );
+
+	// Fedora 26 openssl has no this cipher suites.
+	// QVERIFY( cipherList.contains("TLS_RSA_WITH_RC4_128_SHA") );
+	// QVERIFY( cipherList.contains("TLS_RSA_WITH_RC4_128_MD5") );
+	// QVERIFY( cipherList.contains("SSL_RSA_WITH_RC4_128_SHA") );
+	// QVERIFY( cipherList.contains("SSL_RSA_WITH_RC4_128_MD5") );
 
 	// Fedora 20 openssl has no this cipher suites.
 	// I just believe that F20 has the most strict patent rules
@@ -95,8 +99,6 @@ void TLSUnitTest::testCipherList()
 	QVERIFY( cipherList.contains("SSL_DHE_RSA_WITH_AES_128_CBC_SHA") );
 	QVERIFY( cipherList.contains("SSL_DHE_DSS_WITH_AES_128_CBC_SHA") );
 	QVERIFY( cipherList.contains("SSL_RSA_WITH_AES_128_CBC_SHA") );
-	QVERIFY( cipherList.contains("SSL_RSA_WITH_RC4_128_SHA") );
-	QVERIFY( cipherList.contains("SSL_RSA_WITH_RC4_128_MD5") );
 
 	// QVERIFY( cipherList.contains("SSL_DHE_RSA_WITH_DES_CBC_SHA") );
 	// QVERIFY( cipherList.contains("SSL_DHE_DSS_WITH_DES_CBC_SHA") );
-- 
2.20.1

